services/movie_session.py

```
@@ -0,0 +1,55 @@
+from db.models import MovieSession, Movie, CinemaHall
+from django.db.models import QuerySet
+
+
+from datetime import datetime
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    movie = Movie.objects.get(id=movie_id)
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=cinema_hall,
+        movie=movie
+    )
+    return movie_session
+
+
+def get_movies_sessions(session_date: str = None) -> "QuerySet[MovieSession]":
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    else:
+        return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.filter(id=movie_session_id).first()

```

This violates checklist item #3: "Don't use `.filter()` when you need to get one item, use `.get()` instead." The `.get()` method is designed for retrieving a single object and will raise a helpful `DoesNotExist` error if the object is not found, which is better than silently returning `None`.